registry/api/v2/errors.go

```diff
@@ -15,6 +15,7 @@ var (
 		Message: "The operation is unsupported.",
 		Description: `The operation was unsupported due to a missing
 		implementation or invalid set of parameters.`,
+		HTTPStatusCode: http.StatusMethodNotAllowed,
```

Comment:
Yes, I'm 90% OK with this.  Strictly when you serve a http.StatusNotAllowed you are supposed to specify which verbs are allowed, which we don't do in these cases.  Tim Berners-Lee would not be happy.